test/client_auth/main.d

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+
+    Test for loading client auth name/key from a file.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+
+*******************************************************************************/
+
+module test.client_auth.main;
+
+import test.neo.client.Client;
+import swarm.neo.authentication.CredentialsFile;
+import ocean.transition;
+import ocean.core.Test;
+import ocean.sys.ErrnoException;
+import ocean.io.device.File;
+import Path = ocean.io.Path;
+import ocean.io.select.EpollSelectDispatcher;
+import ocean.text.util.StringC;
+import core.sys.posix.unistd;
+import ocean.stdc.posix.stdlib : mkdtemp;
+
+void main ( )
+{
+    // Exception to throw on failure of C functions.
+    auto e = new ErrnoException;
+
+    // Get current working dir.
+    char[4096] old_cwd;
+    e.enforceRetPtr!(getcwd).call(old_cwd.ptr, old_cwd.length);
+
+    // Create a temp dir.
+    auto tmp_path = e.enforceRetPtr!(mkdtemp)
+        .call("/tmp/Dunittest-XXXXXX\0".dup.ptr);
+    scope ( exit )
+    {
+        auto d_tmp_path = StringC.toDString(tmp_path);
+        // Remove all subdirectories and files in the tmp dir
+        Path.remove(Path.collate(d_tmp_path, "*", true));
+        // Remove the tmp dir itself
+        Path.remove(d_tmp_path);
+    }
+
+    // cd to the temp dir.
+    e.enforceRetCode!(chdir).call(tmp_path);
+    scope ( exit )
+    {
+        e.enforceRetCode!(chdir).call(old_cwd.ptr);
+    }
+
+    // Helper function to write a credentials file and instantiate a client to
+    // read it.
+    void loadCredentials ( cstring credentials )
+    {
+        auto file = new File("credentials", File.ReadWriteCreate);
+        file.write(credentials);
+        file.close();
+
+        void connNotifier ( Client.Neo.ConnNotification ) { }
+
+        auto client = new Client(new EpollSelectDispatcher, "credentials",
+            "127.0.0.1", 10_000, &connNotifier);
```

Comment:
You could just create a temporary credentials file here with
```d
mstring credentialsfilename = "/tmp/Dunittest-credentials-XXXXXX\0".dup[0 .. $ - 1];
int fd = e.enforceRet!(mkstemp)(function (int fd) {return fd >= 0;}).call(credentialsfilename.ptr);
scope (exit)
{
    close(fd);
    e.enforceRetCode!(unlink).call(credentialsfilename.ptr);
}
// Write credentials to the file...
// Pass the credentials file name to the client.
auto client = new Client(new EpollSelectDispatcher, credentialsfilename,
    "127.0.0.1", 10_000, &connNotifier);
```
This would avoid changing the directory.